services/cosi/cosi.go

```diff
@@ -41,24 +40,27 @@ type SignatureRequest struct {
 
 // SignatureResponse is what the Cosi service will reply to clients.
 type SignatureResponse struct {
-	Sum       []byte
+	Hash      []byte
 	Signature []byte
 }
 
 // SignatureRequest treats external request to this service.
-func (cs *CoSi) SignatureRequest(si *network.ServerIdentity, req *SignatureRequest) (network.Body, error) {
+func (cs *CoSi) SignatureRequest(req *SignatureRequest) (network.Body, sda.ClientError) {
+	if req.Roster.ID == sda.RosterID(uuid.Nil) {
+		req.Roster.ID = sda.RosterID(uuid.NewV4())
+	}
 	tree := req.Roster.GenerateBinaryTree()
 	tni := cs.NewTreeNodeInstance(tree, tree.Root, cosi.Name)
 	pi, err := cosi.NewProtocol(tni)
 	if err != nil {
-		return nil, errors.New("Couldn't make new protocol: " + err.Error())
+		return nil, sda.NewClientErrorCode(4100, "Couldn't make new protocol: "+err.Error())
```

Comment:
There are two ways to make a `ClientError`: `NewClientError(error)` and `NewClientErrorCode(int, string)`